src/components/features/Product/ProductOrder/index.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from 'react';
+import styled from '@emotion/styled';
+import { ROUTE_PATH } from '@routes/path';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useAuth } from '@context/auth/useAuth';
+import { Button } from '@components/common';
+import QuantitySelector from './QuantitySelector';
+
+interface ProductOrderProps {
+  name?: string;
+  giftOrderLimit?: number;
+}
+
+export default function ProductOrder({ name, giftOrderLimit }: ProductOrderProps) {
+  const [count, setCount] = useState<number>(0);
+  const { isAuthenticated } = useAuth();
+  const navigate = useNavigate();
+  const { productId } = useParams<{ productId: string }>();
+
+  const handleOrderClick = () => {
+    if (productId) {
+      const orderHistory = { id: Number(productId), count };
+      sessionStorage.setItem('orderHistory', JSON.stringify(orderHistory));
+      const targetPath = isAuthenticated ? ROUTE_PATH.ORDER : ROUTE_PATH.LOGIN;
+      navigate(targetPath);
+    }
+  };
+
+  return (
+    <ProductOrderContainer>
+      <QuantitySelectorConatiner>
+        <Title>{name}</Title>
+        <QuantitySelector giftOrderLimit={giftOrderLimit} onSetCount={setCount} count={count} />
```

Comment:
```suggestion
        <QuantitySelector giftOrderLimit={giftOrderLimit} onSetCount={setCount} />
```

count는 존재하지 않는 속성이라고 타입 에러가 발생하고 있네요.